pubsub/query/query.go

```diff
@@ -152,7 +152,13 @@ func match(tag string, op operator, operand reflect.Value, tags map[string]inter
 	case reflect.Float64:
 		v, ok := value.(float64)
 		if !ok {
-			return false
+			// if value is int, convert it to float64
```

Comment:
Nice, I think this may fix the json encoding issue.  Maybe lose precision?  Dunna how the casting works with floats and int.

As you mentioned above, end-to-end test is important, but I guess that belongs in tendermint itself.